source/slang/slang-check-expr.cpp

```diff
@@ -1145,6 +1177,34 @@ namespace Slang
         return nullptr;
     }
 
+    bool SemanticsVisitor::canStructBeUsedAsSelfDifferentialType(AggTypeDecl *aggTypeDecl)
+    {
+        // A struct can be used as its own differential type if all its members are differentiable
+        // and their differential types are the same as the original types. 
+        //
+        bool canBeUsed = true;
+        for (auto member : aggTypeDecl->members)
+        {
+            if (auto varDecl = as<VarDecl>(member))
+            {
+                if (!isTypeDifferentiable(varDecl->getType()))
```

Comment:
This shouldn't be needed, because calling `tryGetDifferentialType()` on a non diffiernetiable type will return nullptr and the logic below should just work?